java/src/main/native/OrtJniUtil.c

```diff
@@ -183,6 +183,37 @@ size_t onnxTypeSize(ONNXTensorElementDataType type) {
     }
 }
 
+JavaTensorTypeShape getTensorTypeShape(JNIEnv * jniEnv, const OrtApi * api, const OrtValue * value) {
```

Comment:
Have you considered returning an error code that can be compared to ORT_OK and, therefore, using some common construct to check for error codes everywhere?
The JavaTensorTypeShape can be a pointer output parameter.

Reply:
I did but then I'd need to allocate memory for the struct using some allocator and free it. I've tried to avoid allocating memory for things that aren't tensors wherever possible, so this stack allocates it. I could put an ORT error code in the struct itself rather than the valid boolean, which is more informative, though the exception has already been thrown in Java with the proper information by the time this call returns.

Reply:
Can you just declare it on the stack and pass address to it?
I am seeing the function returns this struct by value and assigns it to the stack var.

Reply:
Sure.